js/interactive-guides/circuit-breaker/circuit-breaker-callback.js

```diff
@@ -313,34 +313,57 @@ var circuitBreakerCallBack = (function() {
     };
 
     var __listenToEditorForCircuitBreakerAnnotationChanges = function(editor){
-
-        var __showCircuitBreakerInPod = function(){
+        var __listenToContentChanges = function(editorInstance, changes) {
             // Get pod from contentManager
-            var cb = contentManager.getCircuitBreaker(editor.getStepName());
-
+            var cb = contentManager.getCircuitBreaker(editor.getStepName());            
             // Get the parameters from the editor and send to the circuitBreaker
             var content = editor.getEditorContent();
             try{
-            var annotation = content.match(/@CircuitBreaker(.|\n)*\)/g)[0];
-              annotation = annotation.substring(0,annotation.indexOf("public")).trim(); // Get rid of the public Service...
-              var params = annotation.substring(16,annotation.length-1);
-              // params = params.replace('\n','');
-              params = params.replace(/\s/g, ''); // Remove whitespace
-              params = params.split(',');
-              params.forEach(function(element, index){
-                params[index] = element.trim().substring(element.indexOf('=')+1);
-              });
-
-              cb.updateParameters.apply(cb, params);
+                var matchPattern = "public class BankService\\s*{\\s*@CircuitBreaker\\s*\\(((\\s|\\S)*?)\\)*public Service checkBalance";
```

Comment:
Sorry, I was wondering how the last part of the regexp works as there are spaces and newlines before "public Service checkBalance" but \s is not used to match the whitespace. I played around with the regex tester more. With your regexp or even the one that includes \s* before "public Service checkBalance", it could match

public class BankService {

    @CircuitBreaker(abc=3, (def=4)))
    public Service checkBalance

With that, the regexp should be revised to match any character except () in between ( and ): 
"public class BankService {\\s*@CircuitBreaker\\s*\\((([^\\(\\)])*)\\)?\\s*public Service checkBalance"

Reply:
Updated the pull request with the new regex to exclude parenthesis inside the group.